src/OpenLoco/src/Map/Track/TrackData.cpp

```diff
@@ -389,8 +392,602 @@ namespace OpenLoco::World::TrackData
+            .reverseRotation = 0,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 26,
+            .sparkDirection = false },
+        MiscData{
+            .costFactor = 0x25B,
+            .flags = MiscFlags::verySmallCurve | MiscFlags::oneSided,
+            .reverseTrackId = 42,
+            .reverseRotation = 0,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 26,
+            .sparkDirection = false },
+    };
+
+    std::array<std::string, 11> _pieceFlagsToString{
+        "TrackPieceFlags::diagonal",
+        "TrackPieceFlags::largeCurve",
+        "TrackPieceFlags::normalCurve",
+        "TrackPieceFlags::smallCurve",
+        "TrackPieceFlags::verySmallCurve",
+        "TrackPieceFlags::slope",
+        "TrackPieceFlags::steepSlope",
+        "TrackPieceFlags::oneSided",
+        "TrackPieceFlags::slopedCurve",
+        "TrackPieceFlags::sBend",
+        "TrackPieceFlags::junction",
+    };
+
+    std::array<std::string, 10> _miscToString{
+        "MiscFlags::slope",
+        "MiscFlags::steepSlope",
+        "MiscFlags::curveSlope",
+        "MiscFlags::diagonal",
+        "MiscFlags::verySmallCurve",
+        "MiscFlags::smallCurve",
+        "MiscFlags::curve",
+        "MiscFlags::largeCurve",
+        "MiscFlags::sBendCurve",
+        "MiscFlags::oneSided",
+    };
+
+    std::string toMiscString(uint16_t val)
+    {
+        std::string out;
+        for (auto i = 0; i < 10; ++i)
+        {
+            if (val & (1U << i))
+            {
+                if (out.empty())
+                {
+                    out = _miscToString[i];
+                }
+                else
+                {
+                    out += " | " + _miscToString[i];
+                }
+            }
+        }
+        if (out.empty())
+        {
+            out = "MiscFlags::none";
+        }
+        return out;
+    }
+    std::string toTrackPieceString(uint16_t val)
+    {
+        std::string out;
+        for (auto i = 0; i < 11; ++i)
+        {
+            if (val & (1U << i))
+            {
+                if (out.empty())
+                {
+                    out = _pieceFlagsToString[i];
+                }
+                else
+                {
+                    out += " | " + _pieceFlagsToString[i];
+                }
+            }
+        }
+        if (out.empty())
+        {
+            out = "TrackPieceFlags::none";
+        }
+        return out;
+    }
+
+    const MiscData& getTrackMiscData(size_t trackId)
+    {
+        return _miscData[trackId];
+    }
+
     uint16_t getTrackCompatibleFlags(size_t trackId)
     {
+        for (auto i = 0; i < 44; ++i)
```

Comment:
can use `_miscData.size()` instead of `44`

Reply:
yeah this was just temp code in here to generate the data. Will delete it before it ready.